scripts/cmake-configurator/parser/cmake_row.py

```diff
@@ -32,33 +35,33 @@ def __init__(self, raw_row: str, row_number: int):
       """
       row  = raw_row.strip()
       matches: Optional[Match[str]]  = re.match(self.regex_expression, row)
-      self.type: str =  self.NOT_PARSABLE
+      self.type: str =  types.NOT_PARSABLE.value
       self.val: str = ""
```

Comment:
**ISSUE:** why `val` is a string? If can contain a `bool` or anything else and it's not serialized yet. Therefore, this is one of the few cases where you really shouldn't specify a type for the linter, unless there's something like `dynamic`, I don't remember. Also, initialize it to `None`, not to an empty string. Strings should be used only where it's strictly necessary. This is kind of the opposite of what happens in shell scripting (e.g. Bash) where everything is a string :-)

Reply:
Wouldn't initialzing it to `None` cause more edge case handling?

Reply:
It depends on what self.val is supposed to mean. If it's a in-memory representation of something we parsed (so, deserialized), it cannot be a string. It has to be a generic dynamic object. See the comments about the previous issue.

Reply:
Sorry, i got confused. I was thinking about line 38. You're totally right.